cmd/segment-reaper/observer.go

```diff
@@ -127,6 +134,12 @@ func (obsvr *observer) processSegment(ctx context.Context, path metainfo.ScopedP
 		}
 	}
 
+	if obsvr.from != nil && obsvr.from.Before(pointer.CreationDate) {
+		object.skip = true
+	} else if obsvr.to != nil && obsvr.to.After(pointer.CreationDate) {
+		object.skip = true
+	}
```

Comment:
I think that we can have this block with the following one earlier in the function for avoiding calls when they are skipped, however, let's do these changes once we have the test https://github.com/storj/storj/pull/3658 merged so we can ensure that moving these lines don't break anything.

Reply:
ok